File: src/test/java/com/blackcowmoo/moomark/auth/util/AesUtilTest.java
package com.blackcowmoo.moomark.auth.util;

import java.security.InvalidKeyException;
import java.security.NoSuchAlgorithmException;
import javax.crypto.BadPaddingException;
import javax.crypto.IllegalBlockSizeException;
import javax.crypto.NoSuchPaddingException;
import javax.crypto.SecretKey;
import org.assertj.core.api.Assertions;
import org.junit.jupiter.api.Test;

class AesUtilTest {


  @Test
  void encryptAndDecryptTest()
      throws IllegalBlockSizeException, NoSuchPaddingException,
      NoSuchAlgorithmException, BadPaddingException, InvalidKeyException {
    String plainText = "PLAIN TEXT";

    SecretKey aesSecretKey = AesUtil.generateNewKey();
    byte[] encryptData = AesUtil.encrypt(plainText, aesSecretKey);
    String decryptionText = AesUtil.decrypt(encryptData, aesSecretKey);

    Assertions.assertThat(plainText).isEqualTo(decryptionText);
  }
}

File: src/test/kotlin/com/blackcowmoo/moomark/auth/util/AesUtilTest.kt
package com.blackcowmoo.moomark.auth.util

import org.junit.jupiter.api.Assertions.assertArrayEquals
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertNotEquals
import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest

@SpringBootTest
class AesUtilTest {

  @Autowired
  private lateinit var aesUtil: AesUtil

  @Test
  fun `encrypt and decrypt with generated key`() {
    val key = aesUtil.generateNewKey()
    assertNotNull(key)

    val originalText = "Hello, World!"
    val encrypted = aesUtil.encrypt(originalText, key)
    assertNotNull(encrypted)

    val decrypted = aesUtil.decrypt(encrypted, key)
    assertEquals(originalText, decrypted)
  }

  @Test
  fun `encrypt and decrypt with another generated key`() {
    val key = aesUtil.generateNewKey()
    assertNotNull(key)

    val originalText = "Test encryption/decryption"
    val encrypted = aesUtil.encrypt(originalText, key)
    assertNotNull(encrypted)

    val decrypted = aesUtil.decrypt(encrypted, key)
    assertEquals(originalText, decrypted)
  }

  @Test
  fun `encrypt with one key and decrypt with another key should fail`() {
    val key1 = aesUtil.generateNewKey()
    val key2 = aesUtil.generateNewKey()

    assertNotNull(key1)
    assertNotNull(key2)

    val originalText = "Test with different keys"
    val encrypted = aesUtil.encrypt(originalText, key1)

    val decrypted = aesUtil.decrypt(encrypted, key2)
    assertNotEquals(originalText, decrypted)
  }

  @Test
  fun `generateNewKey returns different keys`() {
    val key1 = aesUtil.generateNewKey()
    val key2 = aesUtil.generateNewKey()

    assertNotNull(key1)
    assertNotNull(key2)
  }
}
